transaction-status/src/parse_token/extension/token_metadata.rs

```diff
@@ -0,0 +1,347 @@
+use {
+    super::*,
+    spl_token_metadata_interface::instruction::{
+        Emit, Initialize, RemoveKey, TokenMetadataInstruction, UpdateAuthority, UpdateField,
+    },
+};
+
+pub(in crate::parse_token) fn parse_token_metadata_instruction(
+    instruction: &TokenMetadataInstruction,
+    account_indexes: &[u8],
+    account_keys: &AccountKeys,
+) -> Result<ParsedInstructionEnum, ParseInstructionError> {
+    match instruction {
+        TokenMetadataInstruction::Initialize(metadata) => {
+            check_num_token_accounts(account_indexes, 4)?;
+            let Initialize { name, symbol, uri } = metadata;
+            let value = json!({
+                "metadata": account_keys[account_indexes[0] as usize].to_string(),
+                "updateAuthority": account_keys[account_indexes[1] as usize].to_string(),
+                "mint": account_keys[account_indexes[2] as usize].to_string(),
+                "mintAuthority": account_keys[account_indexes[3] as usize].to_string(),
+                "name": name,
+                "symbol": symbol,
+                "uri": uri,
+            });
+            Ok(ParsedInstructionEnum {
+                instruction_type: "initializeTokenMetadata".to_string(),
+                info: value,
+            })
+        }
+        TokenMetadataInstruction::UpdateField(update) => {
+            check_num_token_accounts(account_indexes, 2)?;
+            let UpdateField { field, value } = update;
+            let value = json!({
+                "metadata": account_keys[account_indexes[0] as usize].to_string(),
+                "updateAuthority": account_keys[account_indexes[1] as usize].to_string(),
+                "field": field,
```

Comment:
Is this where you're using the serde-traits? I couldn't see where else they would be included.

Reply:
Yeah, just this little guy here. If we're going to flatten it I can probably remove the feature flag as well.